src/pet.js

```diff
@@ -0,0 +1,125 @@
+//magic numbers
+const MAX_FITNESS = 10;
+const MIN_HUNGER = 0;
+const HUNGRY = "I am hungry"
+const UNFIT = "I need a walk"
+const BOTH = "I am hungry AND I need a walk"
+const FINE = "I feel great!"
+const DIED = "Your pet is no longer alive :("
+const ERROR = new Error('Your pet is no longer alive :(')
+
+function Pet(name) {
+    this.name = name;
+    this.age = 0;
+    this.hunger = 0;
+    this.fitness = 4;
+    this.children=[]
+  
+
+}
+
+//getter function 
+Pet.prototype = {
+    get isAlive() {
+        return this.age < 30 && this.hunger < 10 && this.fitness > 0;
+    }
+}
+
+
+
+Pet.prototype.growUp = function () {
+    this.age += 1;
+    this.hunger += 5;
+    this.fitness += 3;
+}
+
+Pet.prototype.walk = function () {
+    if (this.fitness + 4 <= MAX_FITNESS) {
+        this.fitness += 4;
+    }
+    else {
+        this.fitness = MAX_FITNESS;
+    }
+}
+
+Pet.prototype.feed = function () {
+    if (!this.isAlive) {
+        throw ERROR;
+    }
+
+    if (this.hunger < 0) {
+        this.hunger = MIN_HUNGER;
+    }
+
+    if (this.hunger - 3 >= MIN_HUNGER) {
+        this.hunger -= 3;
+    }
```

Comment:
because of how you've set up these if statements, if the hunger is more than 0 but less than 3 and you try to feed the pet the hunger doesn't change (when it should be set to 0). To fix this, you can change this to:
```
    if (this.hunger - 3 >= MIN_HUNGER) {
        this.hunger -= 3;
    } else {
        this.hunger = MIN_HUNGER;
    }
```